client/my-sites/domains/domain-management/dns/dns-add-new.jsx

```diff
@@ -137,15 +163,34 @@ class DnsAddNew extends React.Component {
 
 			const normalizedData = getNormalizedData(
 				formState.getAllFieldValues( this.state.fields ),
-				this.props.selectedDomainName
+				selectedDomainName
 			);
 			this.formStateController.resetFields( this.getFieldsForType( this.state.type ) );
 
-			this.props.addDns( this.props.selectedDomainName, normalizedData ).then(
-				() =>
+			if ( recordToEdit ) {
+				this.props.updateDns( selectedDomainName, [ normalizedData ], [ recordToEdit ] ).then(
+					() => {
+						page( domainManagementDns( selectedSiteSlug, selectedDomainName ) );
+						this.props.successNotice( translate( 'The DNS record has been updated.' ), {
+							duration: 5000,
+						} );
+					},
+					( error ) => {
+						this.props.errorNotice(
+							error.message || translate( 'The DNS record has not been updated.' )
```

Comment:
:information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **23** times:
`translate( 'The DNS record has not been added.' )` *ES Score: 8*
See 2 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/update/edit-dns-records)